cmd/utils/executor.go

```diff
@@ -0,0 +1,236 @@
+// Copyright 2021 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"runtime"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/manifoldco/promptui"
+	"github.com/manifoldco/promptui/screenbuf"
+	"github.com/okteto/okteto/pkg/log"
+	"golang.org/x/term"
+)
+
+type ManifestExecutor interface {
+	Execute(command string, env []string) error
+}
+
+type Executor struct {
+	outputMode string
+}
+
+type jsonMessage struct {
+	Level     string `json:"level"`
+	Stage     string `json:"stage"`
+	Message   string `json:"message"`
+	Timestamp int64  `json:"timestamp"`
+}
+
+// NewExecutor returns a new executor
+func NewExecutor(output string) *Executor {
+	if output == "tty" && !IsSupportForTTY() {
+		output = "plain"
+	}
+	return &Executor{
+		outputMode: output,
+	}
+}
+
+func IsSupportForTTY() bool {
+	if runtime.GOOS != "windows" {
+		return isSupportForTTY()
+	}
+	return false
+}
+
+// Execute executes the specified command adding `env` to the execution environment
+func (e *Executor) Execute(command string, env []string) error {
+
+	cmd := exec.Command("bash", "-c", command)
+	cmd.Env = append(os.Environ(), env...)
+
+	var scanner *bufio.Scanner
+
+	if e.outputMode == "tty" {
+		var (
+			reader io.Reader
+			err    error
+		)
+		if runtime.GOOS != "windows" {
```

Comment:
What would happen if someone in windows executes the deploy command which by default is having `tty` option? If I understood it correctly, command wouldn't be executed, right? Probably I'm missing something

Reply:
they will have the same experience as in Linux but they won't have any color from the output

Reply:
But, I mean, If the option is `tty` and the OS is windows, we are not calling to the command execution, right?. We skip the call to `startCommandTTY` and we are not calling to `cmd.Start` because that is done in the else branch. Am I missing something?